doc/Language/subscripts.pod6

```diff
@@ -253,68 +251,74 @@ dimensions>) the subscript is preserved across the slice operation
-lazily generate values, but once it starts generating values lazily, it
-will stop if it generates a value that points to a nonexistent index.
+This behavior, while at first glance may seem unintuitive, is desirable in
+instances when you want to assign a value at an index in which a defined
+value does not currently exist.
 
-    =begin code :skip-test
-    say @letters[0, 2, 4 ... *];     # Every other element of the array.
-    =end code
-
-This feature is more for protection against accidental out-of-memory
-problems than for actual use.  Since some lazy sequences cache their
-results, every time they are used in a truncation, they accumulate one
-more known element.  Things like this should probably be avoided rather
-than used for effect:
+    =begin code
+    my  @letters;
+    say @letters; # OUTPUT: «[]␤»
 
-    =begin code :skip-test
-    my @a = 2, 3 ... *;
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f)␤»
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f (Any))␤»
+    @letters[^10] = 'a'..'z';
+    say @letters; # OUTPUT: «[a b c d e f g h i j]␤»
     =end code
 
-The runaway protection is not perfect.  The indices are eagerly evaluated,
-with the only stop condition being truncation.  This is to provide
-mostly consistent results when there is self-reference/mutation inside
-the indices.  As such, the following will most likely hang until all
-memory has been consumed:
+If you want the resulting slice to only include defined elements, you can
+silently skip the non-defined elements using the L<#:v> adverb.
 
-    =begin code :preamble<my @letters;>
-    @letters[0 xx *];
+    =begin code
+    my  @letters = <a b c d e f>;
+    say @letters[3..7]:v;  # OUTPUT: «(d e f)␤»
     =end code
 
-So, to safely use lazy indices, they should be one-shot things which
-are guaranteed to overrun the array.  The following alternate formulation
-will produce a fully lazy result (but will not truncate):
+The behavior when indexing a collection via lazy subscripts is different than
+when indexing with their eager counterparts. When accessing via a lazy
+subscript, the resulting slice will be truncated.
 
-    =begin code :preamble<my @letters;>
-    my $a = (0 xx *).map({ @letters[$_] }); # "a", "a", "a" ... forever
+    =begin code :preamble<my @letters = <a b c d e f>;>
+    say @letters[lazy 3..7]; # OUTPUT: «(d e f)␤»
+    say @letters[     3..*]; # OUTPUT: «(d e f)␤»
     =end code
 
-If you I<don't> want to specify your slice as a range/sequence but still want
-to silently skip nonexistent elements, you can use the L<#:v> adverb.
+This behavior exists as a precaution to prevent runaway generation of massive,
+potentially infinite C<Lists> and the out-of-memory issues that occur as a
+result.
+
+Behind the scenes, the implementation is first eagerly accessing the collection
+via subscripts that are already L<reified|/language/glossary#index-entry-Reify>.
```

Comment:
*[feedback]*

IMO this paragraph goes into too much detail:

* Most users shouldn't care about these details and shouldn't need to read all this extra info, at least in this document
* This describes how *Rakudo* implemented this feature, while the docs site tries to be implementation-agnostic and only describe what [the spec](https://github.com/perl6/roast/) specifies the behaviour to be. I don't think the spec describes such minute details as when to reify things in this particular scenario.